packages/core/test/financial-templates/optimistic-rewarder/OptimisticRewarder.js

```diff
@@ -0,0 +1,509 @@
+
+    // Deposit tokens
+    await optimisticRewarder.methods
+      .depositRewards(redemptionToken.options.address, toWei("150"))
+      .send({ from: owner });
+
+    // Submit redemption
+    const redemptions = [{ token: redemptionToken.options.address, amount: toWei("100") }];
+    assert(await didContractThrow(optimisticRewarder.methods.dispute(tokenId, redemptions).send({ from: disputer })));
+    await optimisticRewarder.methods.requestRedemption(tokenId, redemptions).send({ from: submitter });
+    await store.methods.setFinalFee(bondToken.options.address, { rawValue: toWei("1") }).send({ from: owner });
+    receipt = await optimisticRewarder.methods.dispute(tokenId, redemptions).send({ from: disputer });
+    await assertEventEmitted(receipt, optimisticRewarder, "Canceled");
+
+    // Balances should match pre-request balances.
+    assert.equal(await bondToken.methods.balanceOf(submitter).call(), totalBond);
+    assert.equal(await bondToken.methods.balanceOf(disputer).call(), totalBond);
+
+    // Cleanup
+    await store.methods.setFinalFee(bondToken.options.address, { rawValue: finalFee }).send({ from: owner });
+    await optimisticRewarder.methods.sync().send({ from: owner });
+  });
+
+  it("Optimistic oracle reverts on request", async function () {
+    // Mint a token using the standard mint method that emits an UpdateToken event.
+    let receipt = await optimisticRewarder.methods.mint(submitter, updateData).send({ from: tokenUpdater });
+    const tokenId = await getTokenId(receipt);
+
+    // Create and mint tokens.
+    const redemptionToken = await ERC20.new("Redemption", "REDEMPTION", 18).send({ from: owner });
+    await mint(redemptionToken, owner, toWei("150"));
+    await redemptionToken.methods.approve(optimisticRewarder.options.address, toWei("150")).send({ from: owner });
+
+    // Deposit tokens
+    await optimisticRewarder.methods
+      .depositRewards(redemptionToken.options.address, toWei("150"))
+      .send({ from: owner });
+
+    // Submit redemption
+    const redemptions = [{ token: redemptionToken.options.address, amount: toWei("100") }];
+    assert(await didContractThrow(optimisticRewarder.methods.dispute(tokenId, redemptions).send({ from: disputer })));
+    await optimisticRewarder.methods.requestRedemption(tokenId, redemptions).send({ from: submitter });
+    await collateralWhitelist.methods.removeFromWhitelist(bondToken.options.address).send({ from: owner });
+    receipt = await optimisticRewarder.methods.dispute(tokenId, redemptions).send({ from: disputer });
+    await assertEventEmitted(receipt, optimisticRewarder, "Canceled");
+
+    // Balances should match pre-request balances.
+    assert.equal(await bondToken.methods.balanceOf(submitter).call(), totalBond);
+    assert.equal(await bondToken.methods.balanceOf(disputer).call(), totalBond);
+
+    // Cleanup
+    await collateralWhitelist.methods.addToWhitelist(bondToken.options.address).send({ from: owner });
```

Comment:
did you mean to remove this? and the other Cleanup comments on other lines?

Reply:
Nope, the cleanup comment is meant to denote that it's cleaning up the state changes that might interfere with other tests. I can add a comment to that effect or just change the wording. Thoughts?